### PR TITLE
Handle invalid entity ids by showing entity not found pane

### LINKED ISSUE #2289: TypeError: Cannot read property 'id' of undefined
Sentry Issue: [APPSMITH-EW](https://sentry.io/organizations/appsmith/issues/2102336336/?referrer=github_integration)

```
TypeError: Cannot read property 'id' of undefined
  at call (sagas/DatasourcesSagas.ts:362:11)
...
(31 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Added not found page to be displayed when resource does not exist.

Fixes #2289 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/actions/datasourceActions.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const updateDatasource = (payload: Datasource) => {
 };
 
 export const fetchDatasourceStructure = (id: string) => {
+  console.log({ type: ReduxActionTypes.FETCH_DATASOURCE_STRUCTURE_INIT, id });
```
Comment (riodeuno) [inline app/client/src/actions/datasourceActions.ts:20]: missed console.log

File: app/client/src/pages/Editor/PageNotFound.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
```
Comment (riodeuno) [inline app/client/src/pages/Editor/PageNotFound.tsx:1]: Let's rename this to `EntityNotFoundPane`

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Review by riodeuno — DISMISSED** [review]

**Comment by riodeuno** [comment]
Please use the new designs for the Entity not found page. https://www.figma.com/file/a9VOA3LPsaBvDHYxcNs19Z/Generic-Error-State?node-id=0%3A1

File: app/client/src/pages/Editor/EntityNotFoundPane.tsx [hunk 3]
```diff
@@ -28,21 +50,18 @@ const EntityNotFoundPane = () => {
...
-      <div>
-        <p className="bold-text">Entity not found</p>
-        <p>
-          Either this entity doesn&apos;t exist, or you don&apos;t have access
-          to <br />
-          this entity.
+      <div className="page-details">
+        <p className="bold-text">Invalid URL</p>
+        <p className="page-message">
+          The page you’re looking for either does not exist, or cannot be found
```
Comment (riodeuno) [inline app/client/src/pages/Editor/EntityNotFoundPane.tsx:56]: Please use the `messages.ts` file for the copy.

File: app/client/src/pages/Editor/EntityNotFoundPane.tsx [hunk 4]
```diff
@@ -28,21 +50,18 @@ const EntityNotFoundPane = () => {
...
           text="Go Back"
-          intent="primary"
-          icon="arrow-right"
-          iconAlignment="right"
-          size="small"
+          cypressSelector="t--invalid-page-go-back"
           className="button-position"
+          size={Size.large}
+          category={Category.secondary}
           onClick={() => history.goBack()}
```
Comment (riodeuno) [inline app/client/src/pages/Editor/EntityNotFoundPane.tsx:65]: Does a simple `onClick={history.goBack}` work in this scenario?

**Review by riodeuno — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/EntityNotFoundPane.tsx (hunk 4 above)
Comment (vicky-primathon) [reply app/client/src/pages/Editor/EntityNotFoundPane.tsx]: Yes onClick={history.goBack} is working, updated the code

**Review by riodeuno — APPROVED** [review]

**Pull request merged**